### DISCUSSION TITLE
在车牌识别微调的项目中：TypeError: Descriptors cannot not be created directly.

### DISCUSSION THREAD

**Discussion by Boom-Wang** [PaddlePaddle/PaddleOCR#14348] [Q&A] [answered]
### 🔎 Search before asking

- [X] I have searched the PaddleOCR [Docs](https://paddlepaddle.github.io/PaddleOCR/) and found no similar bug report.
- [X] I have searched the PaddleOCR [Issues](https://github.com/PaddlePaddle/PaddleOCR/issues) and found no similar bug report.
- [X] I have searched the PaddleOCR [Discussions](https://github.com/PaddlePaddle/PaddleOCR/discussions) and found no similar bug report.

### 🐛 Bug (问题描述)

TypeError: Descriptors cannot not be created directly.，按道理来说我没有修改任何代码，为什么会出现这个报错？
我已经按照教程一步步执行。
![微信截图_20241207182124](https://github.com/user-attachments/assets/bc667afa-67b4-4cbd-9110-5fb831622873)

### 🏃‍♂️ Environment (运行环境)

本任务基于Aistudio完成, 具体环境如下：

操作系统: Linux
PaddlePaddle: 2.3
paddleslim: 2.2.2
PaddleOCR: Release/2.5

### 🌰 Minimal Reproducible Example (最小可复现问题的Demo)

![2123](https://github.com/user-attachments/assets/d7e1d927-b952-4b83-8893-39dadefee638)

同样，运行下面的微调训练代码也是同样的报错

**Comment by GreatV** [comment]
你提到的问题很可能是由 `protobuf` 版本不兼容引起的。“TypeError: Descriptors cannot be created directly” 这一报错在某些版本的 `protobuf` 与 PaddleOCR 或其依赖项存在兼容性问题时会出现。建议尝试以下步骤来解决：

1. **检查当前 protobuf 版本**：  
   使用 `pip show protobuf` 查看当前安装的 protobuf 版本。

2. **更换 protobuf 版本**：  
   若当前版本过新或不兼容，可尝试安装一个已知稳定的版本，例如：
   ```bash
   pip install --upgrade protobuf==3.20.0
   ```
   
3. **重启环境**：  
   更改 protobuf 版本后，请重启运行环境（如重新启动 AIStudio 的内核），以确保新版本生效。

4. **重新安装相关依赖**

**Comment by Boom-Wang** [comment]
> 你提到的问题很可能是由 `protobuf` 版本不兼容引起的。“TypeError: Descriptors cannot be created directly” 这一报错在某些版本的 `protobuf` 与 PaddleOCR 或其依赖项存在兼容性问题时会出现。建议尝试以下步骤来解决：
> 
> 1. **检查当前 protobuf 版本**：
>    使用 `pip show protobuf` 查看当前安装的 protobuf 版本。
> 2. **更换 protobuf 版本**：
>    若当前版本过新或不兼容，可尝试安装一个已知稳定的版本，例如：
>    ```shell
>    pip install --upgrade protobuf==3.20.0
>    ```
> 3. **重启环境**：
>    更改 protobuf 版本后，请重启运行环境（如重新启动 AIStudio 的内核），以确保新版本生效。
> 4. **重新安装相关依赖**

十分感谢，按照你的建议这个报错已解决，重启AIStudio内核后再次运行出现了新的报错：
![微信截图_20241207233856](https://github.com/user-attachments/assets/69a7633c-d330-4b43-beea-2350f7ceba18)
这个报错我在尝试用ch_PP-OCRv4做微调的时候也出现过，十分苦恼，没找到解决方法

**Comment by Boom-Wang** [comment, marked as answer]
> > 你提到的问题很可能是由 `protobuf` 版本不兼容引起的。“TypeError: Descriptors cannot be created directly” 这一报错在某些版本的 `protobuf` 与 PaddleOCR 或其依赖项存在兼容性问题时会出现。建议尝试以下步骤来解决：
> > 
> > 1. **检查当前 protobuf 版本**：
> >    使用 `pip show protobuf` 查看当前安装的 protobuf 版本。
> > 2. **更换 protobuf 版本**：
> >    若当前版本过新或不兼容，可尝试安装一个已知稳定的版本，例如：
> >    ```shell
> >    pip install --upgrade protobuf==3.20.0
> >    ```
> > 3. **重启环境**：
> >    更改 protobuf 版本后，请重启运行环境（如重新启动 AIStudio 的内核），以确保新版本生效。
> > 4. **重新安装相关依赖**
> 
> 十分感谢，按照你的建议这个报错已解决，重启AIStudio内核后再次运行出现了新的报错： ![微信截图_20241207233856](https://private-user-images.githubusercontent.com/51729645/393528003-69a7633c-d330-4b43-beea-2350f7ceba18.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM2MjQxOTQsIm5iZiI6MTczMzYyMzg5NCwicGF0aCI6Ii81MTcyOTY0NS8zOTM1MjgwMDMtNjlhNzYzM2MtZDMzMC00YjQzLWJlZWEtMjM1MGY3Y2ViYTE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA4VDAyMTEzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzMWE5MzIzZDU0OWQ1Yjg5M2Q2NWFiZjEwMTIxNTU5MDYwNTMyNzAzNmRiMzQ1ODkzOGQ0MTVlM2YwNGVmZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4ElnbsoFo2EelwDygg65oDB-KzxtNfpMbN5SAx5o8oo) 这个报错我在尝试用ch_PP-OCRv4做微调的时候也出现过，十分苦恼，没找到解决方法

已解决，原来是没有按照paddlepaddle-gpu导致的，install后便没有这个报错了
